dcos/dcos-ui
# [DCOS-55016]: node action dropdown, drain active nodes

Closes DCOS-55016

## Testing

1. add this key to the "uiConfiguration" key in Config.dev.ts
```
features: {
  maintenance: true
},
```
2. enable useUIConfigFixtures
3. Visit the nodes page and find the new actions dropdown on the far right (for active nodes)
4. Ensure actions don't appear when feature is disabled
5. Try draining a node (this deactivates the node as well)
6. In order to reactivate the node, issue the following curl command, but replace the Cookie header VALUE and NODE_ID tokens with your own values
```
curl -v -XPOST 'http://localhost:4200/mesos/api/v1' -H 'Content-Type: application/json' -H 'Accept: application/json' --data-binary '{"type":"REACTIVATE_AGENT", "reactivate_agent": {"agent_id": {"value": "NODE_ID"}}}' -H 'Cookie: dcos-acs-auth-cookie=VALUE; dcos-acs-info-cookie=VALUE' --compressed
```

## Notes

I'm told the drained node will eventually be garbage collected by mesos but I'm not sure how long it takes.

## Screenshots

<img width="479" alt="Screen Shot 2019-07-24 at 2 35 36 PM" src="https://user-images.githubusercontent.com/174332/62061978-6a445100-b1e5-11e9-9523-89169c1b4c69.png">

<img width="466" alt="Screen Shot 2019-07-24 at 2 37 08 PM" src="https://user-images.githubusercontent.com/174332/62061981-6c0e1480-b1e5-11e9-9238-b669646c45b9.png">
